src/main/java/org/madhuri/app/message/service/ChannelService.java

```diff
@@ -23,28 +26,37 @@ public void deleteChannel(long id) {
 		channelDAO.deleteChannel(id);
 	}
 
-	
+	public Channel updateChannel(Channel updatedChannel) {
+		if (updatedChannel.getId() == null) {
+			throw new IllegalArgumentException("Channel ID cannot be null");
+		}
+
+		// Check if the channel exists
+		Channel existingChannel = channelDAO.getChannelById(updatedChannel.getId());
+		if (existingChannel == null) {
+			throw new IllegalArgumentException("Channel with ID " + updatedChannel.getId() + " not found");
+		}
+
+		// Update the fields of the existing channel with the updated channel's data
+		existingChannel.setChannelName(updatedChannel.getChannelName());
+		existingChannel.setAdminId(updatedChannel.getAdminId());
+		existingChannel.setWelcomeMessage(updatedChannel.getWelcomeMessage());
+		existingChannel.setUpdatedAt(updatedChannel.getUpdatedAt());
+		existingChannel.setUpdatedBy(updatedChannel.getUpdatedBy());
+		return channelDAO.updateChannel(existingChannel);
```

Comment:
this is called in user service while adding new user to Channel, but the users is not set in existing channel below, 
did we test this code if its working ?